test/price-import.spec.coffee

```diff
@@ -295,3 +295,151 @@ describe 'PriceImport', ->
+      existingProduct =
+        version: 1
+        masterVariant:
+          sku: @sku
+          id: @variantId
+          prices: [ @priceUs ]
+        hasStagedChanges: false
+        published: true
+
+      productToProcess =
+        version: 1
+        masterVariant:
+          sku: @sku
+          id: @variantId
+          prices: [ @priceUs, @priceDe ]
+        hasStagedChanges: false
+        published: true
+
+      @import.publishUpdates = true
+      @import._createOrUpdate([productToProcess], [existingProduct])
+      .then =>
+        actual = updateStub.update.mostRecentCall.args[0]
+        expected =
+          version: productToProcess.version
+          actions: [
+            priceActionDeprecated(
+              "addPrice",
+              @variantId,
+              _.omit(@priceDe, 'id')
+            ),
+            { action: 'publish' }
+          ]
+        expect(actual).toEqual(expected)
+      .catch (err) -> done(err)
+      .finally -> done()
+
+    it 'should not add publish action to update actions when disabled', (done) ->
+      existingProduct =
+        version: 1
+        masterVariant:
+          sku: @sku
+          id: @variantId
+          prices: [ @priceUs ]
+        hasStagedChanges: false
+        published: true
+
+      productToProcess =
+        version: 1
+        masterVariant:
+          sku: @sku
+          id: @variantId
+          prices: [ @priceUs, @priceDe ]
+        hasStagedChanges: false
+        published: true
+
+      @import.publishUpdates = false
+      @import._createOrUpdate([productToProcess], [existingProduct])
+      .then =>
+        actual = updateStub.update.mostRecentCall.args[0]
+        expected =
+          version: productToProcess.version
+          actions: [
+            priceActionDeprecated(
+              "addPrice",
+              @variantId,
+              _.omit(@priceDe, 'id')
+            )
+          ]
+        expect(actual).toEqual(expected)
+      .catch (err) -> done(err)
+      .finally -> done()
+
+    it 'should add publish action to update actions when product not published
+       and has staged changes', (done) ->
+      existingProduct =
+        version: 1
+        masterVariant:
+          sku: @sku
+          id: @variantId
+          prices: [ @priceUs ]
+        hasStagedChanges: true
+        published: false
+
+      productToProcess =
+        version: 1
+        masterVariant:
+          sku: @sku
+          id: @variantId
+          prices: [ @priceUs, @priceDe ]
+        hasStagedChanges: false
+        published: true
+
+      @import.publishUpdates = true
+      @import._createOrUpdate([productToProcess], [existingProduct])
+      .then =>
+        actual = updateStub.update.mostRecentCall.args[0]
+        expected =
+          version: productToProcess.version
+          actions: [
+            priceActionDeprecated(
+              "addPrice",
+              @variantId,
+              _.omit(@priceDe, 'id')
+            )
```

Comment:
sorry the name was not correct.. forgot the 'not' word.
04559526af9fa91061272af241e7951ab72cd192